src/drivers/intel/cavs/interrupt.c

```diff
@@ -31,60 +32,61 @@
 static inline void irq_lvl2_handler(void *data, int level, uint32_t ilxsd,
 				    uint32_t ilxmsd, uint32_t ilxmcd)
 {
-	struct irq_desc *parent = (struct irq_desc *)data;
+	struct irq_desc *parent = data;
 	struct irq_cascade_desc *cascade = container_of(parent,
 						struct irq_cascade_desc, desc);
 	struct irq_desc *child = NULL;
 	struct list_item *clist;
 	uint32_t status;
-	uint32_t i = 0;
 	uint32_t tries = LVL2_MAX_TRIES;
 
 	/* read active interrupt status */
 	status = irq_read(ilxsd);
+	if (!status)
+		return;
 
 	/* handle each child */
-	while (irq_read(ilxsd)) {
+	for (;;) {
+		unsigned int bit = ffs(status) - 1;
+		bool handled = false;
```

Comment:
sorry, why? Where does this requirement come from?..

Reply:
@lyakh It is the style enforced by @lgirdwood and probably comes from Linux kernel itself.

Reply:
@lyakh You won't find it in kernel guidelines, but it's enforced by maintainer and it's better to be consistent. I'm not sure but I guess it's because of c89 compatibilty.